### ISSUE TITLE
CheckExit loop

### ISSUE THREAD

**Issue by Profakos** [SuperSayu/tgstation#175]
This is only called from /turf/Enter

get_turf and this was called after a meteor shower. Could be fake turfs walking around?

**Comment by Profakos** [comment]
The only way to test this is to trigger meteor events for a while, first with faketurfs being created in explosions, and the second by removing them. However, this is not a foolproof test.

Either way, if the loop is probably here? 

```
if(isturf(mover.loc))
    // Nothing but border objects stop you from leaving a tile, only one loop is needed
    for(var/obj/obstacle in mover.loc)
        if(!obstacle.CheckExit(mover, src) && obstacle != mover && obstacle != forget)
            mover.Bump(obstacle, 1)
            return 0
```

Bump() calls Enter again somewhere?

Memo to self: analyse this line https://github.com/SuperSayu/-tg-station/blob/sayustation/code/game/turfs/simulated.dm#L54

**Comment by SuperSayu** [comment]
There are definitely sanity checks that could be added to faketurf/Move() and New(), since faketurfs have a tendency to drag anchored objects with them.  That should never cause an infinite loop... but something might have changed which makes this happen as a side effect.

I will definitely look try to look into it this week.

**Comment by SuperSayu** [comment]
Yeah, it looks like it was caused by meteors being anchored objects, which are treated specially by faketurfs (they try to take anchored objects with them--so for example tables moving when the floor moves, that sort of thing).  Adding sanity checks to faketurfs definitely removes this particular problem, at least I went through a couple rounds of meteors without seeing any of these runtimes.  This will be fixed when I merge the first June PR

**Comment by Profakos** [comment]
The problem might not be specific to meteors then, and anchored objects moving in general at the same time as faketurfs could do it.

Also faketurfs have to be rewritten too, before they can be made again, so that they adhere to the new way turf deconstruction works which also allows the removal of colon operators from them.

**Comment by SuperSayu** [comment]
Fixed in 59b9292fbd95fb86330b28cdf12af5fcca517b45
